fluent.runtime/fluent/runtime/fallback.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+import codecs
+import os
+
+
+class FluentLocalization(object):
+    def __init__(
+        self, locales, resource_ids, resource_loader,
+        use_isolating=False,
+        bundle_class=None, functions=None,
+    ):
+        self.locales = locales
+        self.resource_ids = resource_ids
+        self.resource_loader = resource_loader
+        self.use_isolating = use_isolating
+        if bundle_class is None:
+            from fluent.runtime import FluentBundle
+            self.bundle_class = FluentBundle
+        else:
+            self.bundle_class = bundle_class
+        self.functions = functions
+        self._bundle_cache = []
+        self._bundle_it = self._iterate_bundles()
+
+    def format_value(self, id, args=None):
+        for bundle in self._bundles():
+            if not bundle.has_message(id):
+                continue
+            msg = bundle.get_message(id)
+            if not msg.value:
+                continue
+            val, errors = bundle.format_pattern(msg.value, args)
+            return val
+        return id
+
+    def _bundles(self):
+        bundle_pointer = 0
+        while True:
+            if bundle_pointer == len(self._bundle_cache):
+                try:
+                    self._bundle_cache.append(next(self._bundle_it))
+                except StopIteration:
+                    return
+            yield self._bundle_cache[bundle_pointer]
+            bundle_pointer += 1
+
+    def _iterate_bundles(self):
+        for first_loc in range(0, len(self.locales)):
+            locs = self.locales[first_loc:]
+            for resources in self.resource_loader.resources(locs[0], self.resource_ids):
+                bundle = self.bundle_class(
+                    locs, functions=self.functions, use_isolating=self.use_isolating
+                )
+                for resource in resources:
+                    bundle.add_resource(resource)
+                yield bundle
+
+
+class ResourceLoader(object):
+    def resources(self, locale, resource_ids):
+        raise NotImplementedError
+
+
+class RootedFileResourceLoader(ResourceLoader):
+    def __init__(self, root):
+        self.root = root
+        from fluent.runtime import FluentResource
+        self.Resource = FluentResource
+
+    def resources(self, locale, resource_ids):
+        resources = []
+        for resource_id in resource_ids:
+            path = self.localize_path(os.path.join(self.root, resource_id), locale)
+            if not os.path.isfile(path):
+                continue
+            content = codecs.open(path, 'r', 'utf-8').read()
+            resources.append(self.Resource(content))
+        if resources:
+            yield resources
+
+    def localize_path(self, path, locale):
+        return path.replace("{locale}", locale)
```

Comment:
probably better to use `path.format(locale=locale)` here for more robust string parsing.